Change current Apple OS versions to 26.

bls1999 opened the pull request:
#### a1229133acc9c7252f2ba1b72c76ff542638392f
<pre>
Change current Apple OS versions to 26.
<a href="https://bugs.webkit.org/show_bug.cgi?id=294267">https://bugs.webkit.org/show_bug.cgi?id=294267</a>
<a href="https://rdar.apple.com/152964437">rdar://152964437</a>

Reviewed by Ryan Haddad.

This bumps the current version of all Apple OSes to 26 in webkitpy and the
PrintConfiguration build step on ews-build and build-webkit-org to reflect the
new OS versions Apple announced at WWDC yesterday.

* Tools/CISupport/build-webkit-org/steps.py: Add Tahoe to PrintConfiguration OS name mapping.
* Tools/CISupport/ews-build/steps.py: Ditto.
* Tools/Scripts/webkitpy/common/version_name_map.py: Add Tahoe and bump all current OS versions to 26.
* Tools/Scripts/webkitpy/port/ios.py: Bump current version to 26.
* Tools/Scripts/webkitpy/port/ios_device_unittest.py: Ditto.
* Tools/Scripts/webkitpy/port/ios_simulator_unittest.py: Ditto.
* Tools/Scripts/webkitpy/port/ios_testcase.py: Ditto.
* Tools/Scripts/webkitpy/port/visionos.py: Ditto.
* Tools/Scripts/webkitpy/port/visionos_testcase.py: Ditto.
* Tools/Scripts/webkitpy/port/watch.py: Ditto.
* Tools/Scripts/webkitpy/port/watch_simulator_unittest.py: Ditto.
* Tools/Scripts/webkitpy/port/watch_testcase.py: Ditto.
* Tools/Scripts/webkitpy/port/mac.py: Ditto.
* Tools/Scripts/webkitpy/port/mac_unittest.py:
    (MacTest):
        -&gt; (test_sequoia_baseline_search_path): Renamed to test_latest_baseline_search_path.
        -&gt; (test_downlevel_baseline_search_path): Added new unit test for downlevel baseline search path validation.

Canonical link: <a href="https://commits.webkit.org/296105@main">https://commits.webkit.org/296105@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/5d54082071451813e10abc136eb7de3c40ecbb6a

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/107397 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/131/builds/27079 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/138/builds/17484 "Built successfully") | [  ~~🛠 wpe~~](https://ews-build.webkit.org/#/builders/5/builds/112611 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🛠 win~~](https://ews-build.webkit.org/#/builders/59/builds/57933 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/109361 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/130/builds/27761 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/123/builds/35579 "Built successfully") | [  ~~🧪 wpe-wk2~~](https://ews-build.webkit.org/#/builders/5/builds/112611 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 win-tests~~](https://ews-build.webkit.org/#/builders/59/builds/57933 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/110326 "Passed tests") | [  ~~🧪 ios-wk2~~](https://ews-build.webkit.org/#/builders/132/builds/21995 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 api-mac~~](https://ews-build.webkit.org/#/builders/18/builds/96795 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 api-wpe~~](https://ews-build.webkit.org/#/builders/5/builds/112611 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| [✅ 🧪 webkitpy](https://ews-build.webkit.org/#/builders/6/builds/106802 "Passed tests") | [  ~~🧪 ios-wk2-wpt~~](https://ews-build.webkit.org/#/builders/133/builds/21435 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 mac-wk1~~](https://ews-build.webkit.org/#/builders/135/builds/14925 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🛠 wpe-cairo~~](https://ews-build.webkit.org/#/builders/65/builds/57377 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| | [  ~~🧪 api-ios~~](https://ews-build.webkit.org/#/builders/13/builds/91364 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 mac-wk2~~](https://ews-build.webkit.org/#/builders/136/builds/14959 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/115712 "Built successfully") | 
| | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/128/builds/34464 "Built successfully") | [  ~~🧪 mac-AS-debug-wk2~~](https://ews-build.webkit.org/#/builders/122/builds/25404 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 gtk-wk2~~](https://ews-build.webkit.org/#/builders/1/builds/90569 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| [✅ 🧪 services](https://ews-build.webkit.org/#/builders/28/builds/106905 "Passed tests") | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/121/builds/34839 "Built successfully") | [  ~~🧪 mac-wk2-stress~~](https://ews-build.webkit.org/#/builders/8/builds/93043 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 api-gtk~~](https://ews-build.webkit.org/#/builders/21/builds/90306 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| | [  ~~🧪 vision-wk2~~](https://ews-build.webkit.org/#/builders/126/builds/35219 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 mac-intel-wk2~~](https://ews-build.webkit.org/#/builders/137/builds/13001 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🛠 playstation~~](https://ews-build.webkit.org/#/builders/134/builds/30207 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| [✅ 🛠 🧪 unsafe-merge](https://ews-build.webkit.org/#/builders/22/builds/17369 "Built successfully and passed tests") | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/127/builds/34385 "Built successfully") | | | 
| | [  ~~🛠 tv-sim~~](https://ews-build.webkit.org/#/builders/125/builds/34131 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | | 
| | [  ~~🛠 watch~~](https://ews-build.webkit.org/#/builders/129/builds/37486 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/124/builds/35792 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

gsnedders commented on Tools/CISupport/build-webkit-org/steps.py:
```diff
@@ -2047,9 +2047,9 @@ def convert_build_to_os_name(self, build):
             '15': 'Sequoia',
-            '14': 'Sonoma',
-            '13': 'Ventura',
+            '14': 'Sonoma'
```
Let's keep the trailing comma, which makes the diff here more obvious:

```suggestion
            '14': 'Sonoma',
```

gsnedders commented on Tools/CISupport/ews-build/steps.py:
```diff
@@ -5990,9 +5990,9 @@ def convert_build_to_os_name(self, build):
             '15': 'Sequoia',
-            '14': 'Sonoma',
-            '13': 'Ventura'
+            '14': 'Sonoma'
```
Ditto:

```suggestion
            '14': 'Sonoma',
```

gsnedders commented on Tools/Scripts/webkitpy/common/version_name_map.py:
```diff
@@ -67,11 +67,12 @@ def __init__(self, platform=None):
-            'tvos': self._automap_to_major_version('tvOS', minimum=Version(10), maximum=Version(18)),
-            'watchos': self._automap_to_major_version('watchOS', minimum=Version(1), maximum=Version(11)),
-            'visionos': self._automap_to_major_version('visionOS', minimum=Version(1), maximum=Version(2)),
+            'ios': self._automap_to_major_version('iOS', minimum=Version(10), maximum=Version(26)),
+            'tvos': self._automap_to_major_version('tvOS', minimum=Version(10), maximum=Version(26)),
+            'watchos': self._automap_to_major_version('watchOS', minimum=Version(1), maximum=Version(26)),
+            'visionos': self._automap_to_major_version('visionOS', minimum=Version(1), maximum=Version(26)),
```
Do we actually want to do this? This effectively also defines iOS 20, etc.

I'd have guessed we wanted something more like:

```python
'ios': self._automap_to_major_version('iOS', minimum=Version(10), maximum=Version(18)) + [Version(26)],
```

I do wonder if we should make the API of `_automap_to_major_version` just take an iterator of integers for the version number, so we can do `self._automap_to_major_version('iOS', range(10, 19))` etc. instead, especially if we end up with multiple ranges for each OS.

bls1999 replied:
I considered this when writing the changes. The only real effect here is that we have ios-20 in the baseline search path when testing on iOS versions <= 18 and visionOS versions <= 2. That'll only be relevant for the next 3 months, so I think that's fine.

gsnedders commented on Tools/Scripts/webkitpy/port/ios_device_unittest.py:
```diff
@@ -115,65 +115,65 @@ def test_layout_test_searchpath_with_apple_additions(self):
 
     def test_layout_test_searchpath_without_apple_additions(self):
-        search_path = self.make_port(port_name='ios-device-wk2', os_version=Version(18)).default_baseline_search_path()
+        search_path = self.make_port(port_name='ios-device-wk2', os_version=Version(26)).default_baseline_search_path()
```
Especially when we're testing with a specific `os_version`, having this not equal the default has some value in checking this actually overrides it — and avoids this changing every year.

bls1999 replied:
I can look into making this take from the `CURRENT_VERSION` variable in `ios.py`.

bls1999 replied:
Discussed offline: this will be a follow-up change.

gsnedders commented on Tools/Scripts/webkitpy/port/ios_device_unittest.py:
```diff
@@ -115,65 +115,65 @@ def test_layout_test_searchpath_with_apple_additions(self):
 
     def test_layout_searchpath_wih_device_type(self):
-        search_path = self.make_port(port_name='ios-device-wk2', os_version=Version(18)).default_baseline_search_path(DeviceType.from_string('iPhone SE'))
+        search_path = self.make_port(port_name='ios-device-wk2', os_version=Version(26)).default_baseline_search_path(DeviceType.from_string('iPhone SE'))
```
Likewise, when we're testing with a specific `os_version`, having this not equal the default has some value in checking this actually overrides it — and avoids this changing every year.

bls1999 replied:
Tracking this feedback for both in the conversation above.

gsnedders commented on Tools/Scripts/webkitpy/port/mac_unittest.py:
```diff
@@ -187,11 +187,18 @@ def test_layout_test_searchpath_with_apple_additions(self):
         self.assertEqual(search_path[0], '/mock-checkout/LayoutTests/platform/mac-wk1')
         self.assertEqual(search_path[1], '/mock-checkout/LayoutTests/platform/mac')
 
+    def test_downlevel_baseline_search_path(self):
```
Arguably it's the older downlevel that's the most interesting one, because then we have `mac-sonoma` -> `mac-sequoia` -> `mac`.

bls1999 replied:
Discussed offline -- making 2 new tests to test latest-1 and latest-2.

gsnedders reviewed (changes requested):
The only major comment I have is "does it make sense to change all the tests every year", because testing with a specific `os_version` should presumably stop us from having to update all of that every year, and checks that that argument actually works to override the version.

bls1999 committed:
Change current Apple OS versions to 26.
https://bugs.webkit.org/show_bug.cgi?id=294267
rdar://152964437

Reviewed by Ryan Haddad.

This bumps the current version of all Apple OSes to 26 in webkitpy and the
PrintConfiguration build step on ews-build and build-webkit-org to reflect the
new OS versions Apple announced at WWDC yesterday.

* Tools/CISupport/build-webkit-org/steps.py: Add Tahoe to PrintConfiguration OS name mapping.
* Tools/CISupport/ews-build/steps.py: Ditto.
* Tools/Scripts/webkitpy/common/version_name_map.py: Add Tahoe and bump all current OS versions to 26.
* Tools/Scripts/webkitpy/port/ios.py: Bump current version to 26.
* Tools/Scripts/webkitpy/port/ios_device_unittest.py: Ditto.
* Tools/Scripts/webkitpy/port/ios_simulator_unittest.py: Ditto.
* Tools/Scripts/webkitpy/port/ios_testcase.py: Ditto.
* Tools/Scripts/webkitpy/port/visionos.py: Ditto.
* Tools/Scripts/webkitpy/port/visionos_testcase.py: Ditto.
* Tools/Scripts/webkitpy/port/watch.py: Ditto.
* Tools/Scripts/webkitpy/port/watch_simulator_unittest.py: Ditto.
* Tools/Scripts/webkitpy/port/watch_testcase.py: Ditto.
* Tools/Scripts/webkitpy/port/mac.py: Ditto.
* Tools/Scripts/webkitpy/port/mac_unittest.py:
    (MacTest):
        -> (test_sequoia_baseline_search_path): Renamed to test_latest_baseline_search_path.
        -> (test_downlevel_baseline_search_path): Added new unit test for downlevel baseline search path validation.

Canonical link: https://commits.webkit.org/296105@main
